openassessment/xblock/test/test_submission.py

```diff
@@ -190,6 +191,86 @@ def test_download_url(self, xblock):
         self.assertTrue(resp['success'])
         self.assertEqual(download_url, resp['url'])
 
+
+    def _get_student_item_key(self, num, usage_id):
+        key = "submissions_attachments/test_student/test_course/" + usage_id
+        if num > 0:
+            key = key + "/" + str(num)
+        return key
+
+    def _create_uploaded_file(self, bucket, file_num, usage_id):
+        key = Key(bucket)
+        key.key = self._get_student_item_key(file_num, usage_id)
+        key.set_contents_from_string("How d'ya do?")
+
+    def _create_uploaded_files(self, num_files, usage_key):
+        conn = boto.connect_s3()
+        bucket = conn.create_bucket('mybucket')
+        for i in range(num_files):
+            self._create_uploaded_file(bucket, i, usage_key)
+
+    def _create_entry(self, description, name, size):
+        return {
+            'description': description,
+            'fileName': name,
+            'fileSize': size,
+        }
+
+    @mock_s3
+    @override_settings(
+        AWS_ACCESS_KEY_ID='foobar',
+        AWS_SECRET_ACCESS_KEY='bizbaz',
+        FILE_UPLOAD_STORAGE_BUCKET_NAME="mybucket"
+    )
+    @scenario('data/file_upload_scenario.xml', user_id='bob')
+    def test_delete_and_submit(self, xblock):
+        """ Test that after deleting a file, the remaining files are currectly submitted  """
+        self._create_uploaded_files(5, xblock.scope_ids.usage_id)
+        xblock.xmodule_runtime = Mock(
+            course_id='test_course',
+            anonymous_student_id='test_student',
+        )
+        file_metadata = [
+            self._create_entry('File Number ' + str(i) , 'file_' + str(i), 1000) for i in range(5)
+        ]
+        expected_file_metadata = list(file_metadata)
+        expected_file_metadata.pop(2)
```

Comment:
Nit: create a variable like `file_index_to_delete = 2`, then you can use it here and in line 247 where you make the request to delete the file.